src/handwritten/stream_readers.py

```diff
@@ -238,6 +244,49 @@ def read_one_sample(self, timeout=10):
         """
         return self._interpreter.read_analog_scalar_f64(self._handle, timeout)
 
+    def read_waveform(
+        self, number_of_samples_per_channel=READ_ALL_AVAILABLE, timeout=10
+    ) -> AnalogWaveform[numpy.double]:
+        number_of_samples_per_channel = (
+            self._task._calculate_num_samps_per_chan(
+                number_of_samples_per_channel))
+
+        t0_array = numpy.full(1, numpy.iinfo(numpy.int64).min, dtype=numpy.int64)
+        dt_array = numpy.full(1, numpy.iinfo(numpy.int64).min, dtype=numpy.int64)
+        raw_data = numpy.full(number_of_samples_per_channel, math.inf, dtype=numpy.float64)
```

Comment:
The use of min and inf was for testing. Zeros may be more appropriate for production code.